projects/vdk-core/src/vdk/internal/builtin_plugins/ingestion/ingester_base.py

```diff
@@ -129,6 +130,19 @@ def send_object_for_ingestion(
         """
         See parent doc
         """
+        if self.__object_is_data_frame(payload):
+            column_names = payload.columns.tolist()
+            payload = payload.values.tolist()
```

Comment:
In our codebase it seems we've used `df.itertuples(index=False)` more. https://github.com/search?q=repo%3Avmware%2Fversatile-data-kit+df.itertuples&type=code 

Iterating with itertuples() is generally faster and more memory-efficient than using .values.tolist() as I think it doesn't require converting the entire DataFrame . 

That might be a big pro for this. But we may need to warn people not to modify the data frame after passing it to ingest. Though I don't know if it would be a problem.